tests/unit/test_sae_group.py

```diff
@@ -1,44 +1,90 @@
-from sae_lens.training.sae_group import SAEGroup
+import os
+from pathlib import Path
+
+import torch
+
+from sae_lens.training.sae_group import SparseAutoencoderDictionary
 from tests.unit.helpers import build_sae_cfg
 
 
-def test_SAEGroup_initializes_all_permutations_of_list_params():
+def test_SparseAutoencoderDictionary_initializes_all_permutations_of_list_params():
     cfg = build_sae_cfg(
         d_in=5,
         lr=[0.01, 0.001],
         expansion_factor=[2, 4],
     )
-    sae_group = SAEGroup(cfg)
+    sae_group = SparseAutoencoderDictionary(cfg)
     assert len(sae_group) == 4
-    lr_sae_combos = [(ae.cfg.lr, ae.cfg.d_sae) for ae in sae_group]
+    lr_sae_combos = [(ae.cfg.lr, ae.cfg.d_sae) for _, ae in sae_group]
     assert (0.01, 10) in lr_sae_combos
     assert (0.01, 20) in lr_sae_combos
     assert (0.001, 10) in lr_sae_combos
     assert (0.001, 20) in lr_sae_combos
 
 
-def test_SAEGroup_replaces_layer_with_actual_layer():
+def test_SparseAutoencoderDictionary_replaces_layer_with_actual_layer():
     cfg = build_sae_cfg(
         hook_point="blocks.{layer}.attn.hook_q",
         hook_point_layer=5,
     )
-    sae_group = SAEGroup(cfg)
+    sae_group = SparseAutoencoderDictionary(cfg)
     assert len(sae_group) == 1
-    assert sae_group.autoencoders[0].cfg.hook_point == "blocks.5.attn.hook_q"
+    sae = next(iter(sae_group))[1]
+    assert sae.cfg.hook_point == "blocks.5.attn.hook_q"
 
 
-def test_SAEGroup_train_and_eval():
+def test_SparseAutoencoderDictionary_train_and_eval():
     cfg = build_sae_cfg(
         lr=[0.01, 0.001],
         expansion_factor=[2, 4],
     )
-    sae_group = SAEGroup(cfg)
+    sae_group = SparseAutoencoderDictionary(cfg)
     sae_group.train()
-    for sae in sae_group:
+    for _, sae in sae_group:
         assert sae.training is True
     sae_group.eval()
-    for sae in sae_group:
+    for _, sae in sae_group:
         assert sae.training is False
     sae_group.train()
-    for sae in sae_group:
+    for _, sae in sae_group:
         assert sae.training is True
+
+
+def test_SparseAutoencoderDictionary_save_and_load_model(tmp_path: Path) -> None:
+
+    cfg = build_sae_cfg(
+        lr=[0.01, 0.001],
+        expansion_factor=[2, 4],
+    )
+    sae_group = SparseAutoencoderDictionary(cfg)
+    sae_group.save_saes(f"{tmp_path}/sae_group_saving")
+
+    # Check that the files were saved
+
+    # 2 subfolders in the main folder
+    path = f"{tmp_path}/sae_group_saving"
+    assert len(os.listdir(path)) == 4
+    # each should
+    for subfolder in os.listdir(path):
+        contents = os.listdir(f"{path}/{subfolder}")
+        print(contents)
```

Comment:
nit: looks like this was left in for debugging, but should probably be removed before merging

Reply:
thanks, thats correct.